drivers/video/video_async.c

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "video_async.h"
+#include "video_ctrls.h"
+#include "video_device.h"
+
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/slist.h>
+
+LOG_MODULE_REGISTER(video_async, CONFIG_VIDEO_LOG_LEVEL);
+
+static sys_slist_t notifiers_list;
+
+/* Find a registered notifier associated with this device */
+struct video_notifier *video_async_find_notifier(const struct device *dev)
+{
+	struct video_notifier *nf;
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&notifiers_list, nf, node) {
+		if (nf->dev == dev) {
+			return nf;
+		}
+	}
+
+	return NULL;
+}
```

Comment:
I would be in favor of a `struct video_device_data` to avoid any `notifiers_list`:

```c
struct video_device_data *vdata = dev->data;
struct video_notifier *nf = &vdata->notifier;
struct thisdriver_data *data = vdata->data;
```

Is there anywhere where this would not work?